### DISCUSSION TITLE
Client APIs for administration of Kanidm

### DISCUSSION THREAD

**Discussion by offby0point5** [kanidm/kanidm#3205] [General]
While looking through the path of writing an OpenTofu provider regarding #3155, I found that there is not really any client library for Kanidm. The exception is the pykanidm library which readme says it is to be considered a 4xvery early beta.

What language you wish had a client library for interfacing with Kanidm?
Are you ready to contribute it yourself?

**Comment by offby0point5** [comment]
I do want to look into creating a Golang library of course, so I can use it in the provider.
But I never built or published a Go module before, let alone written any program in Go. So that may take a longer while :sweat_smile:

**Reply by offby0point5** [reply]
After a long time I wanted to come back to this and created a [repo](https://github.com/slop-incubator/go-kanidm) for it.
For sake of getting started, I used LLMs. But further on it is planned to not use 'AI'. Help is appreciated in case you want to 'deslop'.

**Comment by missingfile** [comment]
I suspect the SCIM API will help here. There's probably a Golang library supporting SCIM which means you won't have to write something from scratch.

**Reply by offby0point5** [reply]
It looks to me as if there is not much SCIM in the API yet. But still... I haven't heard of SCIM before, so thank you for the hint to another "nice-to-know" :smile:
